static/app/components/tours/components.tsx

```diff
@@ -0,0 +1,245 @@
+    isAvailable,
+    orderedStepIds,
+    currentStep: null,
+  });
+  const {dispatch, currentStep} = tourContextValue;
+  const isTourActive = currentStep !== null;
+
+  const tourHotkeys = useMemo(() => {
+    return [
+      {match: 'Escape', callback: () => dispatch({type: 'END_TOUR'})},
+      {match: ['left', 'h'], callback: () => dispatch({type: 'PREVIOUS_STEP'})},
+      {match: ['right', 'l'], callback: () => dispatch({type: 'NEXT_STEP'})},
+    ];
+  }, [dispatch]);
+
+  useHotkeys(tourHotkeys);
+
+  return (
+    <tourContext.Provider value={tourContextValue}>
+      {isTourActive && <BlurWindow />}
+      {children}
+    </tourContext.Provider>
+  );
+}
+
+export interface TourElementProps<T extends TourEnumType>
+  extends Partial<UseOverlayProps> {
+  /**
+   * The content being focused during the tour.
+   */
+  children: React.ReactNode;
+  /**
+   * The description of the tour step.
+   */
+  description: React.ReactNode;
+  /**
+   * The unique identifier of the tour step.
+   */
+  id: TourStep<T>['id'];
+  /**
+   * The title of the tour step.
+   */
+  title: React.ReactNode;
+  /**
+   * The relevant tour context
+   */
+  tourContext: TourContextType<T>;
+  /**
+   * The className of the wrapper element.
+   */
+  className?: string;
+}
+
+export function TourElement<T extends TourEnumType>({
+  children,
+  id,
+  title,
+  description,
+  tourContext,
+  position,
+  className,
+}: TourElementProps<T>) {
+  const theme = useTheme();
+
+  const {currentStep, dispatch, orderedStepIds, registerStep} = tourContext;
+  const stepCount = currentStep ? orderedStepIds.indexOf(id) + 1 : 0;
+  const stepTotal = orderedStepIds.length;
+  const hasPreviousStep = stepCount > 1;
+  const hasNextStep = stepCount < stepTotal;
+  const isOpen = currentStep?.id === id;
+
+  const {triggerProps, triggerRef, overlayProps} = useOverlay({
+    isOpen,
+    position,
+    disableTrigger: true,
+  });
+  const element = triggerRef.current;
+
+  useEffect(() => {
+    registerStep({id, element});
+  }, [id, element, registerStep]);
+
+  return (
+    <Fragment>
+      <ElementWrapper
+        className={className}
+        ref={triggerProps.ref}
+        aria-expanded={triggerProps['aria-expanded']}
+      >
+        {children}
+      </ElementWrapper>
+      {isOpen ? (
+        <PositionWrapper zIndex={theme.zIndex.tour.overlay} {...overlayProps}>
+          <TourOverlay animated>
+            <TopRow>
+              <div>
+                {stepCount}/{stepTotal}
+              </div>
+              <TourCloseButton
+                onClick={() => dispatch({type: 'END_TOUR'})}
+                icon={<IconClose style={{color: theme.inverted.textColor}} />}
+                aria-label={t('Close')}
+                borderless
+                size="sm"
+              />
+            </TopRow>
+            <TitleRow>{title}</TitleRow>
+            <div>{description}</div>
+            <Flex justify="flex-end" gap={space(1)}>
```

Comment:
Wait why not use `ButtonBar` for this? It's a bit more semantic, is it just because you need the justify flex end?

Reply:
Pretty much, I'll switch to button bar though :+1: